clientv3/balancer.go

```diff
@@ -221,14 +248,62 @@ func (b *simpleBalancer) updateNotifyLoop() {
 
 func (b *simpleBalancer) notifyAddrs() {
 	b.mu.RLock()
-	addrs := b.addrs
+	eps := make([]addrConn, 0, len(b.addrs))
+	for _, ac := range b.addrs {
+		eps = append(eps, ac)
+	}
+	multi := len(b.addrs) > 1
+	if multi { // sort by state, last failed
+		sort.Slice(eps, func(i, j int) bool {
+			return eps[i].state < eps[j].state || (eps[i].state == eps[j].state && eps[i].lastFailed.Before(eps[j].lastFailed))
+		})
+	}
+	addrs := make([]grpc.Address, 0, len(b.addrs))
+	for _, ac := range eps {
+		addrs = append(addrs, ac.addr)
+	}
 	b.mu.RUnlock()
 	select {
 	case b.notifyCh <- addrs:
 	case <-b.stopc:
 	}
 }
 
+// Up implements gRPC Balancer interface. Client first establishes
+// connections to all given endpoints. As soon as one endpoint is
+// pinned, tear down other connections via 'notifyAddrs', which
```

Comment:
Yeah meant to say if blackholed, since it goes to reset transport loop (tear down, retry blackholed endpoint). I will remove this part and keep the docs much simpler.